src/store/flow/block-types/BaseBlock.ts

```diff
@@ -51,16 +46,22 @@ export const actions = {
   },
 
   /**
-   * This will be the default standard exit mode, but we can override it in the specific block store
+   * This will be the default standard exit mode,
+   * but we can override it in the specific block store (eg: for dynamic test generation in MCQ)
    */
   async handleBranchingTypeChangedToUnified(
-    {dispatch}: {dispatch: Dispatch},
+    {dispatch, rootGetters}: {dispatch: Dispatch, rootGetters: any},
     {block}: {block: IBlock},
   ): Promise<void> {
-    return dispatch('flow/block_updateBranchingExitsWithInvalidScenario', {
-      blockId: block.uuid,
-      test: 'block.value = true',
-    }, {root: true})
+    if (rootGetters['flow/block_shouldHave2Exits'](block.type) === true) {
```

Comment:
The current logic will now handle all blocks'  handleBranchingTypeChangedToUnified, apart from MCQ which could use dynamic test values.
Therefore, I removed all handleBranchingTypeChangedToUnified from each block's store ;)